### DISCUSSION TITLE
FundMe Lesson 3 Sending Eth through a function issue

### DISCUSSION THREAD

**Discussion by rikachet1** [Cyfrin/foundry-full-course-cu#2909] [Q&A] [answered]
During the Lesson 3 FundMe project, I seem to not be able to deploy a contract with ether.

I followed the steps as the video, but when I set 2 Ether as the value then deploy the contract, it says "creation of FundMe errored: Error occurred: revert." 

I've asked ChatGPT and it said to add "constructor() payable {}". This solution allowed me to deploy with 2 Eth, but when I clicked "fund" button/ABI, the above error still occurred.

![Screenshot from 2024-11-22 23-07-03](https://github.com/user-attachments/assets/58be682a-10ab-4e4e-89f9-d421b7fee080)

Here's the FundMe Contract script:
```
contract FundMe {    
    //payable - contracts can hold money like a wallet
    function fund() public payable {
        //Allow users to send $
        //Have a minimum $ sent
        // 1. How do we send ETH to this contract
        //require forces users to pay a minimum of 1 eth for this transaction
        require(msg.value > 1e18, "didn't send enough Eth"); //1e18 = 1 ETH = 1000000000000000000 wei = 1 * 1 ** 18
    }
   //function withdraw() public {}
}
```

**Comment by tcrin** [comment, marked as answer]
Hi @rikachet1,

**Value (Remix IDE): Interacting with Deployed Contracts**  
(You can refer to the guide on how to use Remix IDE [here](https://remix-ide.readthedocs.io/en/latest/udapp.html))  

You are setting **Value** to 2 Ether when deploying the contract, which causes the transaction to revert.

**Solution:**  
Set **Value** to 0 Ether when deploying the contract.

**Note:**  
The **Value** field in Remix IDE must meet the following prerequisites:  
- The contract must be deployed.  
- The function being called in the contract must include the `payable` keyword if it is intended to receive Ether.  
- To avoid revert, ensure that the **Value** aligns with the conditions specified in the `require` statement inside the function.

**Reply by EngrPips** [reply]
Great heads-up.

**Reply by rikachet1** [reply]
Thanks @tcrin !

Setting the Value to 0 Eth allowed me to deploy the contract. 

To add to your answer for anyone else encountering this issue: Once the contract has been deployed, enter "2 Ether" in the Value field and click the "fund" button under the deployed contracts. My mistake was entering 2 Ether in the Value field during deployment, clicking deploy, and then attempting to click "fund."

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.

**Reply by tcrin** [reply]
You're welcome! I'm glad it worked for you 🤗
Thank you for sharing the additional clarification - it’s super helpful for others who might face the same issue. 

It’s great to see the community helping each other out!🚀🚀

**Reply by EngrPips** [reply]
What a great community we have here.

**Reply by saif1163** [reply]
thank you for solving that problem, im also facing that problem but i got my ans here

**Reply by EngrPips** [reply]
Awesome.
